ibm/service/kms/resource_ibm_kms_key_with_policy_overrides.go

```diff
@@ -0,0 +1,306 @@
+				ForceNew:    false,
+				Default:     false,
+			},
+			"crn": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Crn of the key",
+			},
+			"expiration_date": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The date the key material expires. The date format follows RFC 3339. You can set an expiration date on any key on its creation. A key moves into the Deactivated state within one hour past its expiration date, if one is assigned. If you create a key without specifying an expiration date, the key does not expire",
+				ForceNew:    true,
+			},
+			"instance_crn": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Key protect or HPCS instance CRN",
+			},
+			"rotation": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Description: "Specifies the key rotation time interval in months, with a minimum of 1, and a maximum of 12",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "If set to true, Key Protect enables a rotation policy on a single key.",
+						},
+						"interval_month": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validate.ValidateAllowedRangeInt(1, 12),
+							Description:  "Specifies the key rotation time interval in months",
+						},
+					},
+				},
+			},
+			"dual_auth_delete": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Description: "Data associated with the dual authorization delete policy.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "If set to true, Key Protect enables a dual authorization policy on a single key.",
+						},
+					},
+				},
+			},
+			flex.ResourceName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+
+			flex.ResourceCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+
+			flex.ResourceStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the resource",
+			},
+
+			flex.ResourceGroupName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource group name in which resource is provisioned",
+			},
+			flex.ResourceControllerURL: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL of the IBM Cloud dashboard that can be used to explore and view details about the resource",
+			},
+		},
+	}
+}
+
+func resourceIBMKmsKeyWithPolicyOverridesCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Since most part of this function same as create key, is it possible to parameterize the create key function & invoke it here with policies.. this will avoid code duplication.